### PR TITLE
Remove some left-over BSD headers

### PR DESCRIPTION
**Pull request by siying**

### CODE REVIEW

File: db/compaction_iterator.cc [hunk 1]
```diff
@@ -1,5 +1,3 @@
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file. See the AUTHORS file for names of contributors.
```
Comment (sagar0) [inline db/compaction_iterator.cc]: These lines are coming from levelDB isn't it? Shouldn't we let these stay (here and a few other files below)?

Comment (siying) [inline db/compaction_iterator.cc]: But this file is not in LevelDB. I can't go back and track whether any code was copied from there. It wasn't clear to me with a quick peek.

Comment (sagar0) [inline db/compaction_iterator.cc]: ah, ok. :cry:

Comment (siying) [inline db/compaction_iterator.cc]: I took a closer look at the content of the file. I think they are a total rewrite (which is much much more complicated than initial implementation). No code was copied from previous LevelDB code.

Other files were brand new. Not even a rewriting of existing logic.

**Pull request closed without merging**